subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3153,6 +3158,24 @@ void lwm2m_engine_context_init(struct lwm2m_ctx *client_ctx)
 #endif
 }
 
+#if defined(CONFIG_NET_APP_DTLS)
+static int setup_cert(struct net_app_ctx *app_ctx, void *cert)
+{
+#if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
+	struct lwm2m_ctx *client_ctx = CONTAINER_OF(app_ctx,
+						    struct lwm2m_ctx,
+						    net_app_ctx);
+
+	mbedtls_ssl_conf_psk(&app_ctx->tls.mbedtls.conf,
```

Comment:
This function can fail; shouldn't the error bubble up to net_app?

Reply:
Good catch, I'll fix.  This was copy/paste on my part.
FYI: Of the 7 samples that use this, only 2 check it atm.